ui/page/root/home_page.go

```diff
@@ -170,6 +174,193 @@ func (hp *HomePage) OnNavigatedTo() {
+	}
+
+	go func() {
+		hp.AssetsManager.InitializeDEX(hp.dexCtx)
+
+		// If all went well, the dex client must be ready.
+		dexc := hp.AssetsManager.DexClient()
+		if dexc == nil {
+			return // nothing to do
+		}
+
+		// Wait until dex is ready
+		<-dexc.Ready()
+
+		loginDEXNow := dexc.Active()
+		walletsToUnlock := make(map[string]*dexWalletUnlockInfo)
+		for _, xc := range dexc.Exchanges() {
+			for _, bond := range xc.Auth.PendingBonds {
+				loginInfo, err := hp.dexWalletUnlockInfo(bond.AssetID)
+				if err != nil {
+					log.Error(err)
+					continue
+				}
+				walletsToUnlock[bond.Symbol] = loginInfo
+			}
+
+			// Always unlock current bond asset.
+			loginInfo, err := hp.dexWalletUnlockInfo(xc.Auth.BondAssetID)
+			if err != nil {
+				log.Error(err)
+				continue
+			}
+			walletsToUnlock[dex.BipIDSymbol(xc.Auth.BondAssetID)] = loginInfo
+		}
+
+		if !loginDEXNow {
+			loginDEXNow = len(walletsToUnlock) > 0
+		}
+
+		if !loginDEXNow {
+			return // we are done here
+		}
+
+		dexPassEditor := hp.Theme.EditorPassword(new(widget.Editor), values.String(values.StrDexPassword))
+		dexPassEditor.Editor.SingleLine, dexPassEditor.IsRequired = true, true
+
+		// Dex has active trades, login now!
+		loginModal := modal.NewCustomModal(hp.Load).
+			Title(values.String(values.StrLogin)).
+			UseCustomWidget(func(gtx C) D {
+				extra := ""
+				if len(walletsToUnlock) > 0 {
+					extra = values.StringF(values.StrUnlockBondWalletMsg, len(walletsToUnlock))
+				}
+
+				msg := extra
+				if dexc.Active() {
+					msg = values.StringF(values.StrLoginDEXForActiveOrders, extra)
+				}
+
+				return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						return layout.Inset{Bottom: values.MarginPadding10}.Layout(gtx, hp.Theme.Body1(msg).Layout)
+					}),
+					layout.Rigid(func(gtx C) D {
+						// Login dex editor
+						if len(walletsToUnlock) == 0 {
+							return dexPassEditor.Layout(gtx)
+						}
+
+						// Wallet login editor
+						children := []layout.FlexChild{layout.Rigid(dexPassEditor.Layout)}
+						for _, w := range walletsToUnlock {
+							children = append(children, layout.Rigid(func(gtx C) D {
+								return layout.Inset{Top: values.MarginPadding10}.Layout(gtx, w.passEditor.Layout)
+							}))
+						}
+
+						return layout.Inset{Top: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
+							return layout.Flex{Axis: layout.Vertical}.Layout(gtx, children...)
+						})
+					}),
+				)
+			}).
+			SetPositiveButtonText(values.String(values.StrLogin)).
+			SetPositiveButtonCallback(func(isChecked bool, im *modal.InfoModal) bool {
+				dexPassEditor.SetError("")
+
+				// Login to dex first.
+				err := dexc.Login([]byte(dexPassEditor.Editor.Text()))
+				if err != nil {
+					dexPassEditor.SetError(err.Error())
+					return false
+				}
+
+				// Login to wallets.
+				for _, w := range walletsToUnlock {
+					w.passEditor.SetError("")
+
+					err := w.wallet.UnlockWallet(w.passEditor.Editor.Text())
```

Comment:
Same question as above. Would this error if the wallet was already unlocked?

also, I think unlock wallets first so dexc finds ready-to-use wallets when you do dexc login.

Reply:
> Would this error if the wallet was already unlocked?

No, it won't.